java-client/src/main/java/org/opensearch/client/transport/httpclient5/ApacheHttpClient5TransportBuilder.java

```diff
@@ -327,14 +341,19 @@ public static ApacheHttpClient5TransportBuilder builder(HttpHost... hosts) {
 
     private CloseableHttpAsyncClient createHttpClient() {
         // default timeouts are all infinite
-        RequestConfig.Builder requestConfigBuilder = RequestConfig.custom()
-            .setConnectTimeout(Timeout.ofMilliseconds(DEFAULT_CONNECT_TIMEOUT_MILLIS))
-            .setResponseTimeout(Timeout.ofMilliseconds(DEFAULT_RESPONSE_TIMEOUT_MILLIS));
+        RequestConfig.Builder requestConfigBuilder = RequestConfig.custom();
 
         if (requestConfigCallback != null) {
             requestConfigBuilder = requestConfigCallback.customizeRequestConfig(requestConfigBuilder);
         }
 
+        // default timeouts are all infinite
+        ConnectionConfig.Builder connectionConfigBuilder = ConnectionConfig.custom();
```

Comment:
Please add defaults:

```suggestion
        ConnectionConfig.Builder connectionConfigBuilder = ConnectionConfig.custom()
            .setConnectTimeout(Timeout.ofMilliseconds(DEFAULT_CONNECT_TIMEOUT_MILLIS));
```

The builder values could be overridden by `connectionConfigBuilder`

Reply:
Correct, we can add default values in the ConnectionConfigBuilder. These values can be overridden by RequestConfigCallback or ConnectionConfigCallback. Done.